### DISCUSSION TITLE
Adguard Home on Openwrt crashes.

### DISCUSSION THREAD

**Discussion by coronelac** [AdguardTeam/AdGuardHome#5272] [Q&A]
I've noticed my internet going out almost everyday after installing AdGuard. I have confirmed I can get it working by running the command that starts Adguard in verbose, no reboot required.

A couple times I couldn't even reach the router itself.

Log:

[Adguard openwrt.txt](https://github.com/AdguardTeam/AdGuardHome/files/10250689/Adguard.openwrt.txt)

**Comment by akademsubotnik** [comment]
AdguardHome on OpenWRT also problematic for me.  '/etc/init.d/adguardhome status' reports the service as running, but the adguardhome webbrowser gui is not accessible.  Restarting the adguardhome service resolves the problem.  But would like to avoid restarting

**Comment by akademsubotnik** [comment]
AdguardHome had been working perfect for past 3 days.  Today it crashed, last line in log: '2023/06/17 10:55:58.242581 1272#1 [fatal] listen tcp 192.168.1.1:8080: bind: cannot assign requested address'

**Comment by snothub** [comment]
I'm also having very similar issue.

I have AGH running on a secondary router configured as dumb access point (just DNS, no DHCP), and if I reboot that box, system log will show following:
```
Fri Oct 27 21:32:54 2023 daemon.err AdGuardHome[1668]: 2023/10/27 19:32:54.743455 [info] go to http://192.168.88.74:8181
Fri Oct 27 21:32:54 2023 daemon.err AdGuardHome[1668]: 2023/10/27 19:32:54.776110 [info] stopped
Fri Oct 27 21:32:54 2023 daemon.err AdGuardHome[1668]: 2023/10/27 19:32:54.778541 [fatal] listen tcp 192.168.88.74:8181: bind: cannot assign requested address
```

I've tried following to no success:
- change port (I know it's the only thing on that port)
- change start priority (by altering `/etc/rc.d/S25adguardhome` and setting it to 60 or 96)
- adding `service adguardhome start` or `/etc/init.d/adguardhome start` to Local Startup

All of these attempts above gave the same outcome. But starting manually after boot (either in UI or SSH) works fine, service starts up normally.

Edit: seems like statistics get reset after I start up manually, not sure why. 

Hope someone can shed some light on this, looks like a bug in boot sequence.

**Comment by essys** [comment]
> Edit: seems like statistics get reset after I start up manually, not sure why.

This is the standard behavior. You need to mount a persistent working directory, because /var is a symlink to /tmp which naturally gets flushed during reboot.
Please follow up in the OpenWRT wiki:

- https://openwrt.org/docs/guide-user/services/dns/adguard-home#querystatistics_logging
_" The default working directory is /var/adguardhome (By default /var is a symlink to /tmp)."_
_"One of the main benefits of AGH is the detailed query and statistics data provided, however for many routers having long retention periods for this data can cause issues (see flash/storage space requirements). If you are using the default tmpfs storage, you should set a relatively short retention period or disable logging altogether. If you want to have longer retention periods for query/statistics data, consider moving the storage directory to outside your routers flash space."_

- https://openwrt.org/docs/guide-user/services/dns/adguard-home#installation
_"The default configured working directory will mean query logs and statistics will be lost on a reboot. To avoid this you should configure a persistent storage path such as /opt or /mnt with external storage and update the working directory accordingly."_

> Hope someone can shed some light on this, looks like a bug in boot sequence.

I also had some issues with the adguardhome service to autostart, even though it was started and enabled multiple times before.
So I went the easy route and just added the startup cmd in OpenWRT settings:
**System -> Startup -> Local Startup:**
#Autostart AdGuard Home Service
service adguardhome enable
service adguardhome start

![Screenshot from 2024-01-07 14-22-00](https://github.com/AdguardTeam/AdGuardHome/assets/7117619/3b42c9dc-e244-4b87-b5de-74cf16c377ba)
